eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs

```diff
@@ -0,0 +1,805 @@
+        scriptIntegrityHash,
+        adHash,
+        txnetworkid
+      ),
+    inputs',
+    collateral',
+    collateralReturn',
+    outputs',
+    certs',
+    wdrls',
+    txfee',
+    vldt',
+    update',
+    reqSignerHashes',
+    mint',
+    scriptIntegrityHash',
+    adHash',
+    txnetworkid',
+    AlonzoBody,
+    EraIndependentScriptIntegrity,
+    ScriptIntegrityHash,
+  )
+where
+
+import Cardano.Binary
+  ( DecoderError (..),
+    FromCBOR (..),
+    ToCBOR (..),
+    decodeBreakOr,
+    decodeListLenOrIndef,
+    encodeListLen,
+  )
+import Cardano.Crypto.Hash
+import Cardano.Ledger.Address (Addr (..))
+import Cardano.Ledger.Alonzo.Data (AuxiliaryDataHash (..), Data, DataHash, hashData)
+import Cardano.Ledger.Alonzo.TxBody (decodeAddress28, decodeDataHash32, encodeAddress28, encodeDataHash32, getAdaOnly)
+import Cardano.Ledger.BaseTypes
+  ( Network (..),
+    StrictMaybe (..),
+    isSNothing,
+  )
+import Cardano.Ledger.Coin (Coin (..))
+import Cardano.Ledger.Compactible
+import Cardano.Ledger.Core (PParamsDelta)
+import qualified Cardano.Ledger.Core as Core
+import Cardano.Ledger.Credential (Credential (..), StakeReference (..))
+import qualified Cardano.Ledger.Crypto as CC
+import Cardano.Ledger.Era (Crypto, Era)
+import Cardano.Ledger.Hashes
+  ( EraIndependentScriptIntegrity,
+    EraIndependentTxBody,
+  )
+import Cardano.Ledger.Keys (KeyHash (..), KeyRole (..))
+import Cardano.Ledger.Mary.Value (Value (..), policies, policyID)
+import qualified Cardano.Ledger.Mary.Value as Mary
+import Cardano.Ledger.SafeHash
+  ( HashAnnotated,
+    SafeHash,
+    SafeToHash,
+  )
+import Cardano.Ledger.Shelley.CompactAddr (CompactAddr, compactAddr, decompactAddr)
+import Cardano.Ledger.Shelley.Delegation.Certificates (DCert)
+import Cardano.Ledger.Shelley.PParams (Update)
+import Cardano.Ledger.Shelley.Scripts (ScriptHash (..))
+import Cardano.Ledger.Shelley.TxBody (Wdrl (Wdrl), unWdrl)
+import Cardano.Ledger.ShelleyMA.Timelocks (ValidityInterval (..))
+import Cardano.Ledger.TxIn (TxIn (..))
+import Cardano.Ledger.Val
+  ( DecodeNonNegative,
+    Val (..),
+    decodeMint,
+    decodeNonNegative,
+    encodeMint,
+    isZero,
+  )
+import Data.Coders
+import Data.Maybe (fromMaybe)
+import Data.MemoBytes (Mem, MemoBytes (..), memoBytes)
+import Data.Sequence.Strict (StrictSeq)
+import qualified Data.Sequence.Strict as StrictSeq
+import Data.Set (Set)
+import qualified Data.Set as Set
+import Data.Typeable (Proxy (..), Typeable, (:~:) (Refl))
+import Data.Word
+import GHC.Generics (Generic)
+import GHC.Records (HasField (..))
+import GHC.Stack (HasCallStack)
+import GHC.TypeLits
+import NoThunks.Class (InspectHeapNamed (..), NoThunks)
+import Prelude hiding (lookup)
+
+data TxOut era
+  = TxOutCompact
+      {-# UNPACK #-} !(CompactAddr (Crypto era))
+      !(CompactForm (Core.Value era))
+  | TxOutCompactDH
+      {-# UNPACK #-} !(CompactAddr (Crypto era))
+      !(CompactForm (Core.Value era))
+      !(DataHash (Crypto era))
```

Comment:
Note that Alonzo's TxOut has separate pattern synonym defined for `TxOutCompact` and `TxOutCompactDH`. It is crucial not export these type contractors because they allow to bypass the optimization of unpacking ada only cases

Reply:
this is the current pattern synonym for `TxOutCompact`
```hs
-- TODO deprecate
pattern TxOutCompact ::
  ( Era era,
    HasCallStack
  ) =>
  CompactAddr (Crypto era) ->
  CompactForm (Core.Value era) ->
  TxOut era
pattern TxOutCompact addr vl <-
  (viewCompactTxOut -> (addr, vl, SNothing))
  where
    TxOutCompact cAddr cVal = TxOut (decompactAddr cAddr) (fromCompact cVal) SNothing
```
the `TODO deprecate` comment and viewpattern lead me to believe 1. the pattern synonym is mostly for convenience 2. unrolling this pattern synonym will behave identically 3. the intention is that these pattern synonyms will eventually be removed

Reply:
i'll just revert it to be more like it is in alonzo for now, but i'm still curious what the motivation for keeping these is